modules/wsl-distro.nix

```diff
@@ -88,9 +88,13 @@ with lib; {
             copy-launchers = mkIf cfg.startMenuLaunchers (
               stringAfter [ ] ''
                 for x in applications icons; do
-                  echo "Copying /usr/share/$x"
-                  mkdir -p /usr/share/$x
-                  ${pkgs.rsync}/bin/rsync -ar --delete $systemConfig/sw/share/$x/. /usr/share/$x
+                  echo "setting up /usr/share/''${x}..."
+                  if [[ -d $systemConfig/sw/share/$x ]]; then
+                    mkdir -p /usr/share/$x
+                    ${pkgs.rsync}/bin/rsync -ar --delete $systemConfig/sw/share/$x/. /usr/share/$x
+                  else
+                    rm -rf /usr/share/$x
```

Comment:
I am not sure if this would help things or make them worse

Reply:
The directory doesn't exist at all on regular NixOS, that's why I altered the script to remove it. I don't know how Windows will react to that though (I mean, it works even if the directories don't exist, the start menu launchers just don't get created then). I haven't been able to find out how to make WSL re-scan for launchers, but in theory, it would be nice if we were able to make it do that as part of the activation script. Do you think leaving the directory in place and just deleting the files inside would be better?

Reply:
if we want to cleanup the directory we should clean it completely before or delete any file we do not know about. Right now we would miss files usually.

Reply:
The `--delete` option on the rsync command deletes everything in the target directory that doesn't exist at the source